assignment07/fundamental.js

```diff
@@ -0,0 +1,244 @@
+//  Write a function mapBy to convert an array of objects into an object mapped by the specified key:
+// Example: 
+
+
+
+
+let users = [{
+    "id": 1,
+    "first_name": "Nicki",
+    "email": "ncrozier0@squarespace.com",
+    "date_of_birth": "2009/05/09"
+}, {
+    "id": 2,
+    "first_name": "Raychel",
+    "email": "rmcgrady1@cpanel.net",
+    "date_of_birth": "1996/11/05"
+}];
+
+console.log(mapBy(users, "first_name"))
+
+
+function mapBy(users,key){
+    return users.reduce((prev,obj)=>{
+        prev[obj.first_name] = obj
```

Comment:
You are not using the 'key' prop here. Due to this function would map by 'first_name' always